corehq/apps/users/views/__init__.py

```diff
@@ -458,6 +464,16 @@ def invitations(self):
 
     @property
     def page_context(self):
+        if (not self.can_restrict_access_by_location
+            and any(not role.permissions.access_all_locations
+                    for role in self.user_roles)
+        ):
+            messages.warning(self.request, _(
+                "This project has user roles that restrict data access by "
+                "organization, but the software plan no longer supports that. "
+                "Any users assigned to roles that are restricted in data access "
+                "by organization can no longer access this project.  Please "
+                "update the existing roles."))
```

Comment:
Hey Sravan - the checkbox should be set to "True" and disabled when you go back to the UI.  This means we're forcing the user to turn off the feature. 

Since the user is manually turning off the feature for a given role, we shouldn;t re-enable it after upgrade. 

@esoergel 
I'm finding it a little weird that checkbox=True means that the feature is disabled.  Do we want to flip that and change the wording?